viadot/sources/cloud_for_customers.py

```diff
@@ -84,19 +78,31 @@ def _to_records_report(self, url: str) -> List[Dict[str, Any]]:
 
     def _to_records_other(self, url: str) -> List[Dict[str, Any]]:
         records = []
+        tmp_full_url = self.full_url
+        tmp_params = self.params
         while url:
-            response = self.get_response(self.full_url, params=self.params)
+            response = self.get_response(tmp_full_url)
             response_json = response.json()
             if isinstance(response_json["d"], dict):
                 # ODATA v2+ API
                 new_records = response_json["d"].get("results")
+                url = None
+                self.params = None
+                self.endpoint = None
                 url = response_json["d"].get("__next")
+                tmp_full_url = url
+
             else:
                 # ODATA v1
                 new_records = response_json["d"]
+                url = None
+                self.params = None
+                self.endpoint = None
                 url = response_json.get("__next")
+                tmp_full_url = url
 
```

Comment:
This whole bit is hard to understand. Why do we need `self.params` and `self.endpoint` to be set to `None`? And it's also repeated in two places? Also, there is no need to initialize `url` to `None` (eg. line 98) before assigning a value to it. It seems to me like you only need two lines for each case, the one with `new_records` and the one with `url`.

Reply:
Because we run in loop get_response() function which invokes connection, and in a second and higher number of invoking it does not need these parameters anymore. However, after ending these loop is good to set attributes to 'initial' values in terms of invoking to_df(), to_df() many times in jupyter notebook.